docs/resources/gateway.md

```diff
@@ -0,0 +1,61 @@
+---
+page_title: "Equinix Metal: Metal Gateway"
+subcategory: ""
+description: |-
+  Create Equinix Metal Gateways
+---
+
+# metal\_gateway
+
+Use this resource to create Metal Gateway resources in Equinix Metal.
+
+## Example Usage
+
+```hcl
+# Create Metal Gateway for a VLAN with a private IPv4 block with 8 IP addresses
+
+resource "metal_vlan" "test" {
+  description = "test VLAN in SV"
+  metro       = "sv"
+  project_id  = local.project_id
+}
+
+resource "metal_gateway" "test" {
+  project_id               = local.project_id
+  vlan_id                  = metal_vlan.test.id
+  private_ipv4_subnet_size = 8
+}
+```
+
+```hcl
+# Create Metal Gateway for a VLAN and reserved IP address block
+
+resource "metal_vlan" "test" {
+  description = "test VLAN in SV"
+  metro       = "sv"
+  project_id  = local.project_id
+}
+
+resource "metal_reserved_ip_block" "test" {
+  project_id = local.project_id
+  metro      = "sv"
+  quantity   = 2
+}
+
+resource "metal_gateway" "test" {
+  project_id        = local.project_id
+  vlan_id           = metal_vlan.test.id
+  ip_reservation_id = metal_reserved_ip_block.test.id
+}
```

Comment:
I think we would benefit from a guide that demonstrates devices (count = size of address space) being connected to the gateway, with userdata that configures the vlan and an address from the range.

Should we roll that in here as a guide, or another example, or stage this for later with a new issue, @t0mk? (I imagine you may already have tried this and may have the configuration handy).

Reply:
I would document the scerario in a guide. We can roll it in a separate PR I think.